src/main/java/net/dv8tion/jda/core/requests/ratelimit/BotRateLimiter.java

```diff
@@ -311,6 +310,11 @@ public void run()
                         try
                         {
                             request = it.next();
+                            if (!request.runChecks())
+                            {
+                                it.remove();
+                                continue;
```

Comment:
This needs to actually complete the request somehow, not just drop it.
`request.onFailure`.